lib/permanent_records/active_record_5_2.rb

```diff
@@ -0,0 +1,45 @@
+module HandlePermanentRecordsDestroyedInBelongsToAssociation
+  def handle_dependency
+    return unless load_target
+
+    case options[:dependent]
+    when :destroy
+      target.destroy
+      if target.respond_to?(:soft_destroyed?)
```

Comment:
does `permanent_records` define `soft_destroyed`?

Reply:
No, I believe it is our concern that defines this method. But as permanent_records is basically unmaintained I thought it best to cover more possibilities in case our implementation becomes the actual way we do this.

Reply:
it is good to cover it in our case, but in the context of this gem, it is not provided, so thanks for removing it